File: domain/src/main/java/com/crhonvas/domain/model/posts/Post.java
package com.crhonvas.domain.model.posts;

import androidx.annotation.NonNull;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

public class Post {

    private Integer userId;
    private Integer id;
    private String title;
    private String body;

    private Post(@NonNull Builder builder) {
        this.userId = builder.userId;
        this.id = builder.id;
        this.title = builder.title;
        this.body = builder.body;
    }

    public Integer getUserId() {
        return userId == null ? -1 : userId;
    }

    public Integer getId() {
        return id == null ? -1 : id;
    }

    public String getTitle() {
        return title == null ? "" : title;
    }

    public String getBody() {
        return body == null ? "" : body;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (o == null || getClass() != o.getClass()) return false;

        Post post = (Post) o;

        return new EqualsBuilder()
                .append(userId, post.userId)
                .append(id, post.id)
                .append(title, post.title)
                .append(body, post.body)
                .isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(17, 37)
                .append(userId)
                .append(id)
                .append(title)
                .append(body)
                .toHashCode();
    }

    public static final class Builder {
        private Integer userId = -1;
        private Integer id = -1;
        private String title = "";
        private String body = "";

        public Builder() {}

        @NonNull
        public Builder userId(@NonNull Integer userId) {
            this.userId = userId;
            return this;
        }

        @NonNull
        public Builder id(@NonNull Integer id) {
            this.id = id;
            return this;
        }

        @NonNull
        public Builder title(@NonNull String title) {
            this.title = title;
            return this;
        }

        @NonNull
        public Builder body(@NonNull String body) {
            this.body = body;
            return this;
        }

        public Post build() {
            return new Post(this);
        }
    }
}

File: domain/src/main/java/com/crhonvas/domain/model/posts/Post.kt
package com.crhonvas.domain.model.posts

import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder

class Post private constructor(builder: Builder) {
    private val userId: Int?
    private val id: Int?
    private val title: String?
    private val body: String?

    init {
        userId = builder.userId
        id = builder.id
        title = builder.title
        body = builder.body
    }

    fun getUserId(): Int {
        return userId ?: -1
    }

    fun getId(): Int {
        return id ?: -1
    }

    fun getTitle(): String {
        return title ?: ""
    }

    fun getBody(): String {
        return body ?: ""
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val post = o as Post
        return EqualsBuilder()
            .append(userId, post.userId)
            .append(id, post.id)
            .append(title, post.title)
            .append(body, post.body)
            .isEquals
    }

    override fun hashCode(): Int {
        return HashCodeBuilder(17, 37)
            .append(userId)
            .append(id)
            .append(title)
            .append(body)
            .toHashCode()
    }

    class Builder {
        var userId = -1
        var id = -1
        var title = ""
        var body = ""
        fun userId(userId: Int): Builder {
            this.userId = userId
            return this
        }

        fun id(id: Int): Builder {
            this.id = id
            return this
        }

        fun title(title: String): Builder {
            this.title = title
            return this
        }

        fun body(body: String): Builder {
            this.body = body
            return this
        }

        fun build(): Post {
            return Post(this)
        }
    }
}